asm/f64/add_amd64.s

```diff
@@ -0,0 +1,45 @@
+// Copyright ©2016 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// +build !noasm,!appengine
+
+#include "textflag.h"
+
+// func Add(dst, s []float64)
+TEXT ·Add(SB), NOSPLIT, $0
+	MOVQ    dst_base+0(FP), DI
+	MOVQ    dst_len+8(FP), DX
+	MOVQ    s_base+24(FP), SI
+	CMPQ    s_len+32(FP), DX
+	CMOVQLE s_len+32(FP), DX
+	CMPQ    DX, $0
+	JE      add_end
+	XORQ    AX, AX
+	CMPQ    DX, $4
+	JL      add_tail
+
+add_loop:
+	MOVUPS (SI)(AX*8), X0
```

Comment:
MOVUPS moves the same data, the instruction is a byte smaller than MOVUPD, and there's no penalty since it's still the floating point pipeline loading the register.